lib/passport-configurator.js

```diff
@@ -446,7 +446,7 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
               });
             }
           }
-          return res.redirect(successRedirect);
+          return res.redirect(successRedirect + decodeURIComponent(req.query.state || ''));
```

Comment:
You should check the existence of query string of successRedirect before appending the state. For example, if the successRedirect is `http://localhost:3000/success?x=1`, this code will fail to generate the correct url.